src/qat/purr/backends/qblox/live.py

```diff
@@ -103,3 +131,182 @@ def _execute_on_hardware(
                     sweep_iterator.insert_result_at_sweep_position(var_result, response)
 
         return results
+
+
+class NewQbloxLiveEngine(LiveDeviceEngine, InvokerMixin):
+    def startup(self):
+        if self.model.control_hardware is None:
+            raise ValueError(f"Please add a control hardware first!")
+        self.model.control_hardware.connect()
+
+    def shutdown(self):
+        if self.model.control_hardware is None:
+            raise ValueError(f"Please add a control hardware first!")
+        self.model.control_hardware.disconnect()
+
+    def optimize(self, instructions):
+        pass
+
+    def validate(self, instructions: List[Instruction]):
+        pass
+
+    def build_pass_pipeline(self, *args, **kwargs):
+        pipeline = (
+            PassManager()
+            | SweepDecomposition()
+            | RepeatSanitisation()
+            | ScopeSanitisation()
+            | ReturnSanitisation()
+            | ScopeSanitisationValidation()
+            | ReturnSanitisationValidation()
+            | RepeatSanitisationValidation()
+            | TriagePass()
+        )
+        return pipeline
+
+    def _common_execute(self, builder, interrupt: Interrupt = NullInterrupt()):
+        self._model_exists()
+
+        with log_duration("QPU returned results in {} seconds."):
+            res_mgr = ResultManager()
+            self.run_pass_pipeline(builder, res_mgr, self.model)
+            packages = NewQbloxEmitter().emit_packages(builder, res_mgr, self.model)
+            self.model.control_hardware.set_data(packages)
+            playback_results: Dict[str, np.ndarray] = (
+                self.model.control_hardware.start_playback(None, None)
+            )
+
+            # Post execution step needs a lot of work
+            # TODO - Robust batching analysis (as a pass !)
+            # TODO - Lowerability analysis pass
+            # TODO - A generic loop nest model. Sth similar to SweepIterator but does not mixin injection stuff
+
+            triage_result: TriageResult = res_mgr.lookup_by_type(TriageResult)
+            acquire_map = triage_result.acquire_map
+            pp_map = triage_result.pp_map
+            sweeps = triage_result.sweeps
+
+            def create_sweep_iterator():
```

Comment:
Do we need a separate function for this? It's only called once as far as I know.

Reply:
post execution steps are still fuzzy for qblox, will improve on that indeed

Reply:
Happy with this for now as I have greater plans for this part and I will need to recalibrate my reasoning based on internal use cases.